src/kili/utils/labels/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""Set of utils to work with labels."""
+
+from typing import Dict, Generator, Iterable, List, overload
+
+from typing_extensions import Literal
+
+from ...services.json_response.label import ParsedLabel
+
+
+@overload
+def parse_labels(
+    labels: Iterable[Dict], json_interface: Dict, as_generator: Literal[False] = False
+) -> List[ParsedLabel]:
+    ...
+
+
+@overload
+def parse_labels(
+    labels: Iterable[Dict], json_interface: Dict, as_generator: Literal[True]
+) -> Generator[ParsedLabel, None, None]:
+    ...
+
+
+def parse_labels(
+    labels: Iterable[Dict], json_interface: Dict, as_generator: bool = False
```

Comment:
`as_generator` and overloads are not necessary. You can use:
```
LabelIterable = TypeVar('T', bound=Iterable[Dict])
```
and write:
```
def parse_labels(
    labels: LabelIterable, json_interface: Dict) -> LabelIterable
```

Reply:
I've not checked the impact on the user's side typing though.

Reply:
I agree it would be nice to get rid of the as_generator 

But I've tried and it breaks pyright and type hints :/  Not sure how to make it work

Reply:
The return type of this method must be ` -> Iterable[ParsedLabel]`

Reply:
I was able to remove the as_generator, but I can't do it without the `@overload`

Reply:
👌